client/my-sites/test/controller.js

```diff
@@ -0,0 +1,209 @@
+		const initialState = {
+			preferences: {
+				remoteValues: {},
+			},
+			ui: {
+				selectedSiteId: null,
+			},
+		};
+		const store = mockStore( initialState );
+
+		updateRecentSitesPreferences( { store } );
+
+		expect( store.getActions() ).toHaveLength( 0 );
+	} );
+
+	it( 'should not nothing if selected site is the most recent', () => {
+		const selectedSiteId = 1;
+		const initialState = {
+			preferences: {
+				remoteValues: {
+					recentSites: [ selectedSiteId ],
+				},
+			},
+			ui: {
+				selectedSiteId,
+			},
+		};
+		const store = mockStore( initialState );
+
+		updateRecentSitesPreferences( { store } );
+
+		expect( store.getActions() ).toHaveLength( 0 );
+	} );
+
+	it( 'should save the selected site as the most recent', () => {
+		const selectedSiteId = 1;
+		const initialState = {
+			preferences: {
+				remoteValues: {
+					recentSites: [ selectedSiteId + 1 ],
+				},
+			},
+			sites: {
+				items: {
+					[ selectedSiteId ]: {},
+					[ selectedSiteId + 1 ]: {},
+				},
+			},
+			ui: {
+				selectedSiteId,
+			},
+		};
+		const store = mockStore( initialState );
+
+		updateRecentSitesPreferences( { store } );
+
+		expect(
+			store.getActions().find( ( { type } ) => type === PREFERENCES_SET )?.value[ 0 ]
+		).toEqual( selectedSiteId );
+	} );
+
+	it( 'should limit the number of recent sites to 5', () => {
+		const initialState = {
+			preferences: {
+				remoteValues: {
+					recentSites: [ 2, 3, 4, 5, 6 ],
+				},
+			},
+			sites: {
+				items: {
+					1: {},
+					2: {},
+					3: {},
+					4: {},
+					5: {},
+					6: {},
+				},
+			},
+			ui: {
+				selectedSiteId: 1,
+			},
+		};
+		const store = mockStore( initialState );
+
+		updateRecentSitesPreferences( { store } );
+
+		expect(
+			store.getActions().find( ( { type } ) => type === PREFERENCES_SET )?.value
+		).toHaveLength( 5 );
+	} );
+
+	it( 'should not save sites that are not available locally', () => {
+		const selectedSiteId = 1;
+		const initialState = {
+			preferences: {
+				remoteValues: {
+					recentSites: [ selectedSiteId + 1 ],
+				},
+			},
+			sites: {
+				items: {
+					[ selectedSiteId ]: {},
+				},
+			},
+			ui: {
+				selectedSiteId,
+			},
+		};
+		const store = mockStore( initialState );
+
+		updateRecentSitesPreferences( { store } );
+
+		expect(
+			store.getActions().find( ( { type } ) => type === PREFERENCES_SET )?.value
+		).toHaveLength( 1 );
+	} );
+} );
+
+describe( 'recordNoVisibleSitesPageView', () => {
+	it( 'should record the page view', () => {
+		const path = '/path';
+		const siteFragment = 'site';
+		const title = 'Title';
+
+		pageView.recordPageView = jest.fn();
+
+		recordNoVisibleSitesPageView( { path: `${ path }/${ siteFragment }` }, siteFragment, title );
+
+		expect( pageView.recordPageView ).toHaveBeenCalledWith(
+			`/no-sites`,
+			expect.stringMatching( 'All Sites Hidden' ),
+			{ base_path: path }
+		);
+
+		pageView.recordPageView.mockRestore();
```

Comment:
FWIW you can use `jest.spyOn` rather than directly assigning to the function in this way.

Reply:
That's better, thanks!